client/src/main.cpp

```diff
@@ -1,37 +1,32 @@
 #include "logger.h"
 #include "main.h"
 
+#include <algorithm>
+
 bool MyApp::OnInit()
 {
   if (!wxApp::OnInit()) return false;
 
   init_log();
-  LOG(INFO) << "Initializing";
 
-  frame = new MainWindow(wxT("Hello World"), wxPoint(50, 50), wxSize(300, 400));
+  world = World::Generate(100, 11, 5);
+
+  frame = std::make_shared<MainWindow>(
+    wxT("Hello World"), world, wxPoint(50, 50), wxSize(300, 400));
   return true;
```

Comment:
I'm getting crashes on application exit. When I compile the application using Clang and `-fsanitize=address`, I get the following output on crash.
```
==3766==ERROR: AddressSanitizer: attempting free on address which was not malloc()-ed: 0x619000045190 in thread T0
    #0 0x7f2dc967662f in operator delete(void*) (/usr/lib/libasan.so.1+0x5862f)
    #1 0x505263 in MainWindow::~MainWindow() (/media/e/dev/sch/para/CS5500/build/torus+0x505263)
    #2 0x7f2dc8904f71 in wxAppConsoleBase::DeletePendingObjects() (/usr/lib/libwx_baseu-3.0.so.0+0x6ff71)
    #3 0x7f2dc8904ff0 in wxAppConsoleBase::ProcessIdle() (/usr/lib/libwx_baseu-3.0.so.0+0x6fff0)
    #4 0x7f2dc90c19e2 in wxAppBase::ProcessIdle() (/usr/lib/libwx_gtk2u_core-3.0.so.0+0x3709e2)
    #5 0x7f2dc8fc7aad in wxApp::DoIdle() (/usr/lib/libwx_gtk2u_core-3.0.so.0+0x276aad)
    #6 0x7f2dc8fc7be2 (/usr/lib/libwx_gtk2u_core-3.0.so.0+0x276be2)
    #7 0x7f2dc4c3162c in g_main_context_dispatch (/usr/lib/libglib-2.0.so.0+0x4962c)
    #8 0x7f2dc4c31a07 (/usr/lib/libglib-2.0.so.0+0x49a07)
    #9 0x7f2dc4c31d31 in g_main_loop_run (/usr/lib/libglib-2.0.so.0+0x49d31)
    #10 0x7f2dc5edb466 in gtk_main (/usr/lib/libgtk-x11-2.0.so.0+0x12f466)
    #11 0x7f2dc8fe91d4 in wxGUIEventLoop::DoRun() (/usr/lib/libwx_gtk2u_core-3.0.so.0+0x2981d4)
    #12 0x7f2dc894ad4f in wxEventLoopBase::Run() (/usr/lib/libwx_baseu-3.0.so.0+0xb5d4f)
    #13 0x7f2dc8906f05 in wxAppConsoleBase::MainLoop() (/usr/lib/libwx_baseu-3.0.so.0+0x71f05)
    #14 0x7f2dc89a281c in wxEntry(int&, wchar_t**) (/usr/lib/libwx_baseu-3.0.so.0+0x10d81c)
    #15 0x4d020a in main /home/mogwa/dev/sch/para/CS5500/client/src/main.cpp:32
    #16 0x7f2dc66127ff in __libc_start_main (/usr/lib/libc.so.6+0x207ff)
    #17 0x4cf5b8 in _start (/media/e/dev/sch/para/CS5500/build/torus+0x4cf5b8)

0x619000045190 is located 16 bytes inside of 936-byte region [0x619000045180,0x619000045528)
allocated by thread T0 here:
    #0 0x7f2dc967612f in operator new(unsigned long) (/usr/lib/libasan.so.1+0x5812f)
    #1 0x4fae2d in __gnu_cxx::new_allocator<std::_Sp_counted_ptr_inplace<MainWindow, std::allocator<MainWindow>, (__gnu_cxx::_Lock_policy)2> >::allocate(unsigned long, void const*) (/media/e/dev/sch/para/CS5500/build/torus+0x4fae2d)
    #2 0x4fa3f1 in std::allocator_traits<std::allocator<std::_Sp_counted_ptr_inplace<MainWindow, std::allocator<MainWindow>, (__gnu_cxx::_Lock_policy)2> > >::allocate(std::allocator<std::_Sp_counted_ptr_inplace<MainWindow, std::allocator<MainWindow>, (__gnu_cxx::_Lock_policy)2> >&, unsigned long) (/media/e/dev/sch/para/CS5500/build/torus+0x4fa3f1)
    #3 0x4f91f2 in std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count<MainWindow, std::allocator<MainWindow>, wchar_t const (&) [12], std::shared_ptr<World>&, wxPoint, wxSize>(std::_Sp_make_shared_tag, MainWindow*, std::allocator<MainWindow> const&, wchar_t const (&) [12], std::shared_ptr<World>&, wxPoint&&, wxSize&&) (/media/e/dev/sch/para/CS5500/build/torus+0x4f91f2)
    #4 0x4f70e5 in std::__shared_ptr<MainWindow, (__gnu_cxx::_Lock_policy)2>::__shared_ptr<std::allocator<MainWindow>, wchar_t const (&) [12], std::shared_ptr<World>&, wxPoint, wxSize>(std::_Sp_make_shared_tag, std::allocator<MainWindow> const&, wchar_t const (&) [12], std::shared_ptr<World>&, wxPoint&&, wxSize&&) (/media/e/dev/sch/para/CS5500/build/torus+0x4f70e5)
    #5 0x4f22f1 in std::shared_ptr<MainWindow>::shared_ptr<std::allocator<MainWindow>, wchar_t const (&) [12], std::shared_ptr<World>&, wxPoint, wxSize>(std::_Sp_make_shared_tag, std::allocator<MainWindow> const&, wchar_t const (&) [12], std::shared_ptr<World>&, wxPoint&&, wxSize&&) (/media/e/dev/sch/para/CS5500/build/torus+0x4f22f1)
    #6 0x4eb087 in std::shared_ptr<MainWindow> std::allocate_shared<MainWindow, std::allocator<MainWindow>, wchar_t const (&) [12], std::shared_ptr<World>&, wxPoint, wxSize>(std::allocator<MainWindow> const&, wchar_t const (&) [12], std::shared_ptr<World>&, wxPoint&&, wxSize&&) (/media/e/dev/sch/para/CS5500/build/torus+0x4eb087)
    #7 0x4e665a in _ZSt11make_sharedI10MainWindowIRA12_KwRSt10shared_ptrI5WorldE7wxPoint6wxSizeEES4_IT_EDpOT0_ (/media/e/dev/sch/para/CS5500/build/torus+0x4e665a)
    #8 0x4d00c1 in MyApp::OnInit() /home/mogwa/dev/sch/para/CS5500/client/src/main.cpp:14
    #9 0x4df359 in wxAppConsoleBase::CallOnInit() (/media/e/dev/sch/para/CS5500/build/torus+0x4df359)
    #10 0x7f2dc89a2841 in wxEntry(int&, wchar_t**) (/usr/lib/libwx_baseu-3.0.so.0+0x10d841)
    #11 0x7f2dc66127ff in __libc_start_main (/usr/lib/libc.so.6+0x207ff)
```

I'm pretty sure `wxFrame`s need to instantiate as normal pointers, otherwise wxWidgets tries to delete it on application exit. Not sure how it does that though.